src/sql/workbench/services/objectExplorer/browser/asyncServerTree.ts

```diff
@@ -106,7 +106,8 @@ export class AsyncServerTree extends WorkbenchAsyncDataTree<ConnectionProfileGro
 		while (stack.length > 0) {
 			const node = stack.pop();
 			if (node) {
-				if (!this.isCollapsed(node.element)) {
+				// Root node is always expanded and isCollapsed method does not work for it
```

Comment:
The root node isn't always expanded though. Could you explain more what the issue is and how this is fixing it?

Reply:
Or are you referring to the "root" group that we have?

Reply:
I am referring to the root of the tree that is always expanded. The this.isCollapsed method throws an error for the tree root.

Reply:
Please update the comment to specifically explain this - the whole "secret root group" thing we have isn't obvious to anyone not intimately familiar with how the OE tree is constructed so having the context here will avoid confusion in the future - especially since this is apparently an "edge case" thing that doesn't affect normal functionality. 

Did you look into why it throws an error? That doesn't seem like it should be happening in the first place.

Reply:
The error happens because someone down the code the tree looks in the list of visible elements and the root not is not present there which throws an error.

Reply:
I will update the comment and include that information.

Reply:
Thanks, in the future please try to include this kind of information in the initial PR description/comments - it helps reviewers to understand the context and ensure we're fixing the right thing. 

(taking a look at removing this "root" node thing in the future once we've gotten rid of the old tree seems like it could be useful anyways. Don't think we need it  with the new tree since it already has the concept of the root "node" with the use of undefined when getting children)